### ISSUE TITLE
Performance tests compilation failure on Raspberri Pi 3

### ISSUE THREAD

**Issue by ihhub** [ihhub/penguinV#138]
Original post from from @0x72D0 in #132 

device: raspberry 3B
cpu: cortex-a53
linux kernel: 4.9.35-v7+

ok I succeed to compile it with your suggestion, but it doesn't work with the -march=native. It give me the following error (after the make):

![2018-10-12-210645_1824x984_scrot](https://user-images.githubusercontent.com/16087328/46894826-5487ba00-ce65-11e8-8f3c-3cd20fc97c97.png)

but with -mcpu=cortex-a53 and -mfpu=neon-vfpv4 everything work fine.

**Comment by ihhub** [comment]
`-march` flag is for compilation of specific parts of code related to SIMD instructions. As I can see from your log compilation fails on blob detection which does not have any SIMD specific code. Could you please try to compile an example with blob detection which locates in `examples/blob_detection` folder.  This example does not have such flag.

**Comment by 0x72D0** [comment]
well it fail when I compile blob_detection for the first time, after removing the -march=native in the Makefile it works.

![2018-10-13-152554_1824x984_scrot](https://user-images.githubusercontent.com/16087328/46907067-96177400-cefc-11e8-88a2-69fa82bd41e0.png)

**Comment by ihhub** [comment]
Seems like `-march=native` causes problems in many other projects as well. Some people suggest to try `-mcpu=native`

**Comment by 0x72D0** [comment]
-mcpu=native give me the same result...

**Comment by ihhub** [comment]
I'll try to check it myself today or tomorrow.

**Comment by ihhub** [comment]
I got a chance to test the compilation on Raspberry Pi 3B today. `-mfpu=neon-vfpv4` instead of `-march=native` is sufficient. With just this flag I have no issue of compiling unit and performance tests.

I ran performance tests and I could say that we really need SIMD implementation as timings for normal non-multithreaded functions are really high.

**Comment by 0x72D0** [comment]
yup it's works fine with `-mfpu=neon-vfpv4`, so maybe we can make the makefile to detect automatically when we are on arm and call g++ with `-mfpu=neon-vfpv4` instead of `-march=native`
